src/Forms/Form.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from "react";
+import produce from "immer";
+import PersonalDetails from "./PersonalDetails";
+import Dates from "./Dates";
+import Payment from "./Payment";
+
+class Form extends Component {
+  state = {
+    formControls: {
+      startDate: {
+        value: "2018-07-01",
+        valid: true,
+        touched: false,
+        errorMessage: "start date is requried"
+      },
+      endDate: {
+        value: "2018-07-05",
+        valid: true,
+        touched: false,
+        errorMessage: "end date is required"
+      },
+      firstName: {
+        value: "",
+        valid: false,
+        touched: false,
+        errorMessage: "first name is required"
+      },
+      lastName: {
+        value: "",
+        valid: false,
+        touched: false,
+        errorMessage: "last name is required"
+      },
+      email: {
+        value: "",
+        valid: false,
+        touched: false,
+        errorMessage: "email is required"
+      },
+      gender: {
+        value: "",
+        valid: false,
+        touched: false,
+        errorMessage: "gender is required"
```

Comment:
The shape of these objects looks very similar. Do you think we can create this initial state structure more succinctly using `reduce` somehow?

Reply:
startDate and endDate have default values and valid is true. The rest of them don't have defaults and valid is false. The error messages are similar to the field name... but the field name is sometimes camelcase combined 2 words...

Reply:
I would probably start out with 
```const fields = ['startDate', 'endDate', 'firstName', 'lastName', 'email', 'gender']```
and transform it into 
```{
    startDate: {
      value: "2018-07-01",
      valid: true,
      touched: false,
      errorMessage: "start date is requried"
    },
    endDate: {
      value: "2018-07-05",
      valid: true,
      touched: false,
      errorMessage: "end date is required"
    },
    firstName: {
      value: "",
      valid: false,
      touched: false,
      errorMessage: "first name is required"
    },
    lastName: {
      value: "",
      valid: false,
      touched: false,
      errorMessage: "last name is required"
    },
    email: {
      value: "",
      valid: false,
      touched: false,
      errorMessage: "email is required"
    },
    gender: {
      value: "",
      valid: false,
      touched: false,
      errorMessage: "gender is required"
    }
  }```

Reply:
Would the use of reduce() be a case of over-engineering?